addon/serializers/active-model-serializer.js

```diff
@@ -38,19 +38,49 @@ export default Serializer.extend({
   normalize(payload) {
     let type = Object.keys(payload)[0];
     let attrs = payload[type];
+    let { belongsToAssociations, hasManyAssociations } = this.schema._registry[type].class.prototype;
+    let belongsToKeys = Object.keys(belongsToAssociations);
+    let hasManyKeys = Object.keys(hasManyAssociations);
 
     let jsonApiPayload = {
       data: {
         type: pluralize(type),
-        attributes: {}
+        attributes: {},
+        relationships: {}
       }
     };
     if (attrs.id) {
       jsonApiPayload.data.id = attrs.id;
     }
+
     Object.keys(attrs).forEach((key) => {
       if (key !== 'id') {
-        jsonApiPayload.data.attributes[dasherize(key)] = attrs[key];
+        if (this.serializeIds === 'always') {
+          if (belongsToKeys.includes(key)) {
+            let association = belongsToAssociations.hasOwnProperty(key);
```

Comment:
Your tests only cover if it throws not if records are actually persisted in server database.